chain-storage/src/api.rs

```diff
@@ -80,6 +86,10 @@ pub fn store_chain_state<T: StoredChainState>(
         (COL_APP_HASHS, encoded_height.clone()),
         genesis_state.get_last_app_hash().to_vec(),
     );
+    db.set(
+        (COL_STAKING_VERSIONS, encoded_height.clone()),
+        genesis_state.get_staking_version().encode(),
+    );
```

Comment:
should this be under `write_history_states`?

Reply:
or can it function properly without historical states? I assume the use case here would be that:
1. one gets the app hash (that one can verify its block was signed on by expected validators)
2. get components of app hash from historical app state and verify they match the app state
3. fetch state at that height and verify the proof leads to that expected root (from the components)

Reply:
Actually, if the light client doesn't trust the node, it can't just fetch state from the node, it needs to store the root hash it trusted during synchronization, then verify the proof against it.